src/util.ts

```diff
@@ -7,4 +11,15 @@ export function getRandomId(): string {
   rand += String(randomIntFromInterval(1000000, 999999999));
 
   return rand;
-}
\ No newline at end of file
+}
+
+export function decode<T>(decoder: D.Decoder<unknown, T>) {
+  return (input: unknown): T => pipe(decoder.decode(input), fold(
+    error => {
+      // eslint-disable-next-line
+      console.error("Decoding error", error);
+      throw new Error(D.draw(error));
```

Comment:
(possibly for later) You should create custom error (`export class DecodingError implements Error {...}`) so that you can retrieve in the error collector (typically the route handler) using `if (e instanceof DecodingError) {` to return a specific error code, log the message error there, etc.

Reply:
I have created the DecodingError, thanks! A PR specifically about error handling is in my backlog though. I'll work on the route handler and error collection later in this PR :)